scylla-cdc-lib/src/main/java/com/scylladb/cdc/lib/LocalTransport.java

```diff
@@ -72,34 +69,28 @@ public void configureWorkers(Map<TaskId, SortedSet<StreamId>> workerConfiguratio
         currentGenerationId = workerConfigurations.keySet().stream()
                 .findAny().map(TaskId::getGenerationId);
         stop();
-        stopped = false;
-        int wCount = Math.min(workersCount, workerConfigurations.size());
-        workerThreads = new Thread[wCount];
-        Map<TaskId, SortedSet<StreamId>>[] tasks = split(workerConfigurations, wCount);
-        for (int i = 0; i < wCount; ++i) {
-            WorkerConfiguration workerConfiguration =
-                    workerConfigurationBuilder
-                            .withTransport(this)
-                            .withConsumer(new TaskAndRawChangeConsumerAdapter(consumerProvider.getForThread(i)))
-                            .build();
-
-            workerThreads[i] = new WorkerThread(workersThreadGroup, i, workerConfiguration, tasks[i]);
-            workerThreads[i].start();
-        }
-    }
 
-    private static Map<TaskId, SortedSet<StreamId>>[] split(Map<TaskId, SortedSet<StreamId>> tasks, int wCount) {
-        @SuppressWarnings("unchecked")
-        Map<TaskId, SortedSet<StreamId>>[] result = new Map[wCount];
-        for (int i = 0; i < wCount; ++i) {
-            result[i] = new HashMap<>();
-        }
-        int pos = 0;
-        for (Map.Entry<TaskId, SortedSet<StreamId>> e : tasks.entrySet()) {
-            result[pos].put(e.getKey(), e.getValue());
-            pos = (pos + 1) % wCount;
-        }
-        return result;
+        WorkerConfiguration workerConfiguration = workerConfigurationBuilder.withTransport(this)
+                .withConsumer(new TaskAndRawChangeConsumerAdapter(consumerProvider.getForThread(0))).build();
```

Comment:
Previously, each consumer was consuming changes always on the same thread. This is not true any more, is it?

Reply:
No, as stated in doc etc, consumers are now expected to be thread safe. I realize this is a rather big change, but we talked about this, and in general we should encourage (and ours are) consumers to be stateless.